librz/core/cmd_type.c

```diff
@@ -1883,6 +1919,84 @@ RZ_IPI RzCmdStatus rz_type_define_handler(RzCore *core, int argc, const char **a
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_IPI RzCmdStatus rz_type_list_enum_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	const char *enum_name = argc > 1 ? argv[1] : NULL;
+	// TODO: Reconsider the `te <enum_name> <member_value>` syntax change
+	const char *member_value = argc > 2 ? argv[2] : NULL;
+	if (enum_name) {
+		if (member_value) {
+			return types_enum_member_find(core, enum_name, member_value);
+		} else {
+			types_enum_print(core, enum_name, mode);
+		}
+	} else {
+		// A special case, since by default `te` returns only the list of all enums
+		if (mode == RZ_OUTPUT_MODE_STANDARD) {
+			mode = RZ_OUTPUT_MODE_QUIET;
+		}
+		types_enum_print_all(core, mode);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_enum_bitfield_handler(RzCore *core, int argc, const char **argv) {
+	const char *enum_name = argc > 1 ? argv[1] : NULL;
+	const char *enum_member = argc > 2 ? argv[2] : NULL;
+	Sdb *TDB = core->analysis->sdb_types;
+	char *output = rz_type_enum_member(TDB, enum_name, enum_member, 0);
+	if (!output) {
+		eprintf("Cannot find anything matching the specified bitfield");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_println(output);
+	free(output);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_enum_c_handler(RzCore *core, int argc, const char **argv) {
+	const char *enum_name = argc > 1 ? argv[1] : NULL;
+	Sdb *TDB = core->analysis->sdb_types;
+	print_enum_in_c_format(TDB, enum_name, true);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_enum_c_nl_handler(RzCore *core, int argc, const char **argv) {
+	const char *enum_name = argc > 1 ? argv[1] : NULL;
+	Sdb *TDB = core->analysis->sdb_types;
+	print_enum_in_c_format(TDB, enum_name, false);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_enum_find_handler(RzCore *core, int argc, const char **argv) {
+	const char *enum_value = argc > 1 ? argv[1] : NULL;
+	return types_enum_member_find_all(core, enum_value);
+}
+
+RZ_IPI RzCmdStatus rz_type_list_function_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	const char *function = argc > 1 ? argv[1] : NULL;
+	Sdb *TDB = core->analysis->sdb_types;
+	if (function) {
+		// A special case, since by default `tf name` returns the output in SDB format
+		if (mode == RZ_OUTPUT_MODE_STANDARD) {
+			mode = RZ_OUTPUT_MODE_SDB;
+		}
+		types_function_print(core, function, mode);
+	} else {
+		if (mode == RZ_OUTPUT_MODE_JSON) {
+			print_keys(TDB, core, stdiffunc, printfunc_json_cb, false);
+		} else {
+			print_keys(TDB, core, stdiffunc, printkey_cb, false);
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_function_c_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
```

Comment:
in oldshell i'm not able to make `tfcj` work. Also, what would be the difference between `tfcj` and `tcj` in this new implementation? I think they are the same thing. Honestly, I can't think about a "C JSON" reprensetation. IMHO this should be merged with `tf`, as said above.

Reply:
Removed it.